ruby/sdk-lightweight.md

```diff
@@ -0,0 +1,194 @@
+# Ruby SDK (lightweight overview)
+
+This is a lightweight proposal for the Ruby SDK. The intention for this proposal is to give everyone
+an end-to-end feel for the Ruby SDK. Many aspects of this proposal will probably change in the more
+detailed versions later.
+
+
+## Types
+
+After many experiments we've decided to not proceed with Ruby typing at this point. This decision
+will be revisited when Ruby's typing system matures to support non-basic use-cases.
+
+
+## Defining an activity
+
+Activities are defined as classes ([a very common practise](https://guides.rubyonrails.org/active_job_basics.html)
+given that the language is object-oriented).
+
+`sum_activity.rb`
+
+```ruby
+class SumActivity < Temporal::Activity
+  # This will generate a default class-level stub SumActivity.execute to be used in workflows
+  def execute(a, b)
+    a + b
+  end
+end
+```
+
+#### Alternative approaches
+
+- Using plain methods as activities. This is not feasible in Ruby since non-namespaced methods
+  are global without a clear way to reference them (would have to be `method(:my_activity)`)
+
+- It is possible to allow multiple activities per class, for example:
+
+```ruby
+class CalculatorActivity < Temporal::Activity
+  # Generate custom stubs to be used in workflows:
+  # - CalculatorActivity.add
+  # - CalculatorActivity.subtract
+  activities :add, :subtract
+
+  def add(a, b)
+    a + b
+  end
+
+  def subtract(a, b)
+    a - b
+  end
+end
+```
+
+
+## Defining a workflow
+
+Similar to activities, workflows are also represented as classes.
+
+`sum_workflow.rb`
+
+```ruby
+require 'sum_activity'
+
+class SumWorkflow < Temporal::Workflow
+  def execute(a, b)
+    # - .execute! is a shorthand for .execute().await
+    # - only positional arguments are supported (inline with other languages), keyword
+    #   arguments are reserved for options
+    SumActivity.execute(a, b, start_to_close_timeout: 10)
```

Comment:
Can I see an example of starting multiple activities and waiting on the first to finish? Just curious how this looks in Ruby.